File: Assets/Scripts/GameData.js
#pragma strict

public static var moves: int = 0;

public function ResetMoves() {
	moves = 0;
}

public static var lockRotate: boolean = false;
public var cameraRotater: GameObject;

public function ToggleRotateLock() {
	lockRotate = !lockRotate;
	if(lockRotate) {
		cameraRotater.GetComponent.<LerpToTransform>().Reset();
	}
}

public function ResetRotateLock() {
	lockRotate = false;
}


public static var lockInput: boolean = false;

public function ToggleInputLock() {
	lockInput = !lockInput;
}

public function ResetInputLock() {
	lockInput = false;
}

File: Assets/Scripts/GameData.cs
using UnityEngine;
using System.Collections;

public class GameData : MonoBehaviour {

	public static int moves = 0;

	public void  ResetMoves (){
		moves = 0;
	}

	public static bool  lockRotate = false;
	public GameObject cameraRotater;

	public void  ToggleRotateLock (){
		lockRotate = !lockRotate;
		if(lockRotate) {
			cameraRotater.GetComponent<LerpToTransform>().Reset();
		}
	}

	public void  ResetRotateLock (){
		lockRotate = false;
	}


	public static bool  lockInput = false;

	public void  ToggleInputLock (){
		lockInput = !lockInput;
	}

	public void  ResetInputLock (){
		lockInput = false;
	}
}